delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/AclsRoutes.scala

```diff
@@ -0,0 +1,277 @@
+        } yield Project(orgLabel, projLabel)).fold(
+          err => reject(validationRejection(err.getMessage)),
+          provide(_)
+        )
+      case _                                                                                                      => reject
+
+    }
+
+  private def extractAclAddressFilter: Directive1[AclAddressFilter] =
+    (extractUnmatchedPath & parameter("ancestors" ? false)).tflatMap { case (path, ancestors) =>
+      path match {
+        case Path.Slash(Path.Segment(`any`, Path.Empty))                                  => provide(AnyOrganization(ancestors))
+        case Path.Slash(Path.Segment(`any`, Path.Slash(Path.Segment(`any`, Path.Empty)))) =>
+          provide(AnyOrganizationAnyProject(ancestors))
+        case Path.Slash(Path.Segment(org, Path.Slash(Path.Segment(`any`, Path.Empty))))   =>
+          Label(org).fold(
+            err => reject(validationRejection(err.getMessage)),
+            label => provide(AnyProject(label, ancestors).asInstanceOf[AclAddressFilter])
+          )
+        case _                                                                            => reject
+      }
+    }
+
+  def routes: Route = baseUriPrefix(baseUri.prefix) {
+    pathPrefix("acls") {
+      extractCaller { implicit caller =>
+        concat(
+          extractAclAddress { address =>
+            extractSubject { implicit subject =>
+              parameter("rev" ? 0L) { rev =>
+                concat(
+                  (put & entity(as[AclInput])) { aclEntity =>
+                    authorizeFor(address, aclsWrite).apply {
+                      val status = if (rev == 0L) Created else OK
+                      completeIO(status, acls.replace(address, aclEntity.toAcl, rev).map(_.void))
+                    }
+                  },
+                  (patch & entity(as[PatchAcl])) { aclPatch =>
+                    authorizeFor(address, aclsWrite).apply {
+                      aclPatch match {
+                        case AppendAcl(acl)   =>
+                          completeIO(acls.append(address, acl, rev).map(_.void))
+                        case SubtractAcl(acl) =>
+                          completeIO(acls.subtract(address, acl, rev).map(_.void))
+                      }
+                    }
+                  },
+                  delete {
+                    authorizeFor(address, aclsWrite).apply {
+                      completeIO(OK, acls.delete(address, rev).map(_.void))
+                    }
+                  },
+                  (get & parameter("self" ? true)) {
+                    case true  =>
+                      (parameter("rev".as[Long].?) & parameter("ancestors" ? false)) {
+                        case (Some(_), true)    => reject(simultaneousRevAndAncestorsRejection)
+                        case (Some(rev), false) =>
+                          completeIO(
+                            acls
+                              .fetchSelfAt(address, rev)
```

Comment:
Shouldn't the operations return a SearchResult so as not having to map it here again ?

Reply:
it is the decision of the routes to convert a `AclResource` to a `SerchResult`, so it seems accurate to me that it is done by the routes.

Reply:
But then it is accurate to have the search context in this operation ?